libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt

```diff
@@ -1273,17 +1273,40 @@ object ArrayOps : TemplateGroupBase() {
     }
 
     val f_fill = fn("fill(element: T, fromIndex: Int = 0, toIndex: Int = size)") {
-        platforms(Platform.JVM)
         include(InvariantArraysOfObjects, ArraysOfPrimitives, ArraysOfUnsigned)
     } builder {
         doc { "Fills original array with the provided value." }
         returns("Unit")
         body {
             """
-            java.util.Arrays.fill(this, fromIndex, toIndex, element)
+            if (fromIndex !in 0..size) {
+              throw IndexOutOfBoundsException("fromIndex ${'$'}fromIndex out of range [0, ${'$'}size]")
+            }
+            if (toIndex > size) {
+              throw IndexOutOfBoundsException("toIndex ${'$'}toIndex out of range [${'$'}fromIndex, ${'$'}size]")
+            }
+            require(fromIndex <= toIndex) { "fromIndex(${'$'}fromIndex) > toIndex(${'$'}toIndex)" }
+            for (index in fromIndex until toIndex) {
+                this[index] = element
+            }
```

Comment:
Given that Kotlin/Native has its own implementation and Kotlin/JS will be delegating to array's method, there seem to be no need in this manual implementation.

Reply:
Oh, ok. Do I point to its implementation somehow? Or remove it from being generated? I don't really understand how native tracks these kind of changes being in a separate repo.

Reply:
Yes, you need to call implementation methods (e.g. https://github.com/JetBrains/kotlin-native/blob/9e26d32/runtime/src/main/kotlin/kotlin/collections/ArrayUtil.kt#L78) here, however building Kotlin/Native together with Kotlin can be quite complicated, so feel free to leave this exercise to maintainers :-)

Reply:
Ok I left it as an `arrayFill` function call, but obviously implementations will need to be added to that file similar to `arrayCopy`.

It's not clear to me whether the responsibility for the argument checks should live in the template for native or in the implementation. If it's in the template, let me know and I'll update.